### ISSUE TITLE
Table block: Placeholder setup screen redesign

### ISSUE THREAD

**Issue by mapk** [WordPress/gutenberg#15820]
Currently the Table block has a setup screen, that doesn't really follow suit to our other placeholder setup screens.

I think it might be more consistent to redesign the setup screen for the Table block to match those of other blocks. Sound good? 

**Table block**

![Screen Shot 2019-05-24 at 12 31 03 PM](https://user-images.githubusercontent.com/617986/58352429-5c0e3900-7e20-11e9-8729-71e61931f42e.png)

**Proposed Table block setup screen** (ignore the toolbar)

![table-placeholder](https://user-images.githubusercontent.com/617986/58353229-cd4eeb80-7e22-11e9-8236-2979a405aec6.png)

This is meant to be a quick iteration to add consistency. From here we can explore other ways in which to improve the setup for a Table block.

**Comment by kjellr** [comment]
This is a great idea. @mapk, what do you think of stacking the fields like so? I think that seems like a more natural way to visually scan these fields:

![Untitled-1](https://user-images.githubusercontent.com/1202812/58435674-96314200-808f-11e9-9dcc-79bb123e8e5a.png)

I'd also like it if we can align the button a little better, but I'm not sure how if we want to keep it centered too.

**Comment by mapk** [comment]
> what do you think of stacking the fields like so?

I like it better. I wasn't sure if labels needed to proceed the number field for accessibility purposes. Let's plan on using your mockups (keeping everything centered works fine) and get some a11y feedback here as well.

**Comment by talldan** [comment]
Planning to look at this with @tellthemachines tomorrow. We'll work on something that looks like @kjellr's design.

> I wasn't sure if labels needed to proceed the number field for accessibility purposes. 

My opinion is that we should stick with labels before, preferably above inputs. A quick look around and I couldn't see anywhere else in Gutenberg that has labels after a number or text input. The design guidelines for the component also have some advice (sorry! 😄)
https://github.com/WordPress/gutenberg/tree/master/packages/components/src/text-control#label-text

> Label text should be above the input field, and always visible.

From an accessibility point of view, there's not much information, but I did find this:
https://www.w3.org/TR/WCAG20-TECHS/H44.html

> For all input elements of type text, file or password, for all textareas and for all select elements in the Web page:
> 1. Check that there is a label element that identifies the purpose of the control before the input, textarea, or select element

**Comment by afercia** [comment]
Thanks for labelling this issue with "Needs Accessibility Feedback" 🙂 

Yes, labels need to proceed the number field as it's the case for all form controls with the exception of checkboxes and radio buttons.

Re: the label text: I wouldn't recommend to change them to "Columns" and "Rows". The current text gives more context and it's clearer.

Aside: language related question: is "count" correct in this case? I'm not a native English speaker, but wouldn't "number" or "amount" be better?
